upgrade/upgrade-scripts/upgrade-container

```diff
@@ -429,14 +429,36 @@ function migrate_file() {
 		cp "$path" "/var/lib/machines/${CONTAINER}${path}" ||
 			die "'cp $path' failed for '$CONTAINER'"
 	else
-		if [[ -f "/var/lib/machines/${CONTAINER}${path}" ]]; then
-			# If the target file doesn't exist in the host then
-			# we delete the file. This case could happen if the host
-			# configuration doesn't exist but a fresh install placed
-			# package default configuration.
-			rm "/var/lib/machines/${CONTAINER}${path}" ||
-				die "failed to remove '$path' in '$CONTAINER'"
-		fi
+		#
+		# If the target file doesn't exist in the host then we
+		# delete the file. This case could happen if the host
+		# configuration doesn't exist but a fresh install placed
+		# package default configuration.
+		#
+		rm -f "/var/lib/machines/${CONTAINER}${path}" ||
+			die "failed to remove file '$path' in '$CONTAINER'"
+	fi
+}
+
+function migrate_dir() {
+	local path="$1"
+	local parentdir
+
+	if [[ -d "$path" ]]; then
+		parentdir="$(dirname "$path")"
+		mkdir -p "/var/lib/machines/${CONTAINER}${parentdir}" ||
+			die "'mkdir -p $parentdir' failed for '$CONTAINER'"
+		cp -rT "$path" "/var/lib/machines/${CONTAINER}${path}" ||
```

Comment:
Do we need to preserve the attributes for these files?

Reply:
I don't know. We're not currently doing it when migrating files, so I didn't bother doing it for these directories.

I'd be OK adding the `-p` option (or w/e the option is) to preserver the attributes if you think that'd be the safest bet; I'm not really sure either way, and not sure how to verify if it's needed or not.